repose-aggregator/src/docs/asciidoc/recipes/rate-limiting.adoc

```diff
@@ -1,72 +1,124 @@
+For additional info on populating that header and on rate limiting by groups, see the <<../filters/rate-limiting.adoc#,Rate Limiting Filter>> documentation.
 
 == Configuration
-* *Default Configuration:* rate-limiting.cfg.xml
-* *Released:* prior to version v1.0.6
-* link:../schemas/rate-limiting-configuration.xsd[Schema]
-
-Rate limiting can be configured to use any of the datastore types to store rate limiting information for users.
-By default, rate limiting will use the distributed datastore (`hash-ring`) if available.
-If no distributed datastores are available, then rate limiting will use the local datastore.
+=== System Model
+The filter can be enabled by adding it to the list of filters in the <<../architecture/system-model.adoc#,System Model>>.
 
 [source,xml]
 .system-model.cfg.xml
 ----
 <?xml version="1.0" encoding="UTF-8"?>
+<!-- To configure Repose see: http://wiki.openrepose.org/display/REPOSE/Configuration -->
 <system-model xmlns="http://docs.openrepose.org/repose/system-model/v2.0">
   <repose-cluster id="repose">
-    <nodes>
-      <node id="node1" hostname="localhost" http-port="9090"/>
-      <node id="node2" hostname="localhost" http-port="8080"/>
-    </nodes>
+    ...
     <filters>
       <filter name="rate-limiting"/>
     </filters>
-    <services>
-      <service name="dist-datastore"/>
-    </services>
-    <destinations>
-      <endpoint id="endpoint" protocol="http" hostname="originservice" root-path="/" port="8080" default="true"/>
-    </destinations>
-  </repose-cluster>
+    ...
 </system-model>
 ----
 
-In the following example, the Repose instance with the `node` `id` of `node1` will launch a Distributed Datastore service which will listen on port 9999.
-The Repose instance with the `node` `id` of `node2` will launch it's own Distributed Datastore service which will listen on port 7777.
-With `allow-all` set to `false`, a host that is not in the cluster cannot communicate with the Distributed Datastore.
+=== Rate Limiting
+After the filter has been added to the <<../architecture/system-model.adoc#,System Model>>, the example configuration can copied/moved from the examples directory to the configuration files directory (i.e. `/etc/repose`).
+The example configuration for rate limiting is sufficient for testing and will limit requests to the origin service to 10 times per minute.
 
 [source,xml]
-.dist-datastore.cfg.xml
+.rate-limiting.cfg.xml
 ----
-<?xml version="1.0" encoding="UTF-8"?>
-<distributed-datastore xmlns='http://docs.openrepose.org/repose/distributed-datastore/v1.0'>
-    <allowed-hosts allow-all="false"/>
-    <port-config>
-        <port port="9999" cluster="repose"/>
-        <port port="7777" cluster="repose" node="node2"/>
-    </port-config>
-</distributed-datastore>
+include::https://raw.githubusercontent.com/rackerlabs/repose/master/repose-aggregator/components/filters/rate-limiting-filter/src/main/resources/META-INF/schema/examples/rate-limiting.cfg.xml[]
```

Comment:
Another spot where we are relying on GitHub. In this case, you're also breaking versioning, since that configuration could differ between master and the current version. I understand that `include` effectively takes a snapshot of what is being included, but consider cutting a bug fix release, or some other release which puts us behind the current version.

Since you are using `include` anyway, why not just `include` the file in the repository directly? That should just be simple file resolution, so you shouldn't have to copy that file around or anything.

Reply:
Good point. Relative pathing hell, here I come.

Reply:
This also obsoletes the `allow-uri-read` attribute.

Reply:
Oh no! I just learned about that, too!

Oh well.